core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/execution/ExecutorServiceProvider.java

```diff
@@ -68,11 +72,24 @@ public ExecutorServiceProvider(int corePoolSize, int maxPoolSize, int queueCapac
             SECONDS,
             queueCapacity > 0 ? new LinkedBlockingQueue<>(queueCapacity) : new SynchronousQueue<>(),
             factory);
-    executor.setRejectedExecutionHandler(
-        (r, __) -> LOG.warn("Executor rejected to handle the payload {}", r));
+    executor.setRejectedExecutionHandler(rejectedExecutionHandler);
     executor.prestartCoreThread();
   }
 
+  /**
+   * @param corePoolSize - corePoolSize of ThreadPoolExecutor
+   * @param maxPoolSize - maximumPoolSize of ThreadPoolExecutor
+   * @param queueCapacity - queue capacity. if > 0 then this is capacity of {@link
+   *     LinkedBlockingQueue} if <=0 then {@link SynchronousQueue} are used.
+   */
+  public ExecutorServiceProvider(int corePoolSize, int maxPoolSize, int queueCapacity) {
+    this(
+        corePoolSize,
+        maxPoolSize,
+        queueCapacity,
+        (r, executor1) -> LOG.warn("Executor rejected to handle the payload {}", r));
```

Comment:
I think `e` as a variable name would be prettier than `executor1`.

Reply:
I think particular case it would be useful to use logger of implementation but not superclass. It would show which executor rejected task (minor, metered minor, etc)